File: mobile/src/main/java/com/siliconlabs/bledemo/utils/FilterDeviceParams.java
package com.siliconlabs.bledemo.Utils;

import com.siliconlabs.bledemo.BeaconUtils.BleFormat;

import java.util.List;

public class FilterDeviceParams {
    private String filterName;
    private String name;
    private String advertising;
    private int rssiValue;
    private boolean rssiFlag;
    private boolean onlyFavourite;
    private boolean onlyConnectable;
    private List<BleFormat> bleFormats;


    public FilterDeviceParams(String filterName, String name, String advertising, int rssiValue, boolean rssiFlag,
                              List<BleFormat> bleFormats, boolean onlyFavourite, boolean onlyConnectable) {
        this.filterName = filterName;
        this.name = name;
        this.advertising = advertising;
        this.rssiValue = rssiValue;
        this.rssiFlag = rssiFlag;
        this.bleFormats = bleFormats;
        this.onlyFavourite = onlyFavourite;
        this.onlyConnectable = onlyConnectable;
    }

    public boolean isEmptyFilter() {
        return (name == null || name.equals(""))
                && (advertising == null || advertising.equals(""))
                && !rssiFlag
                && !onlyFavourite
                && !onlyConnectable
                && (bleFormats == null || bleFormats.isEmpty());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAdvertising() {
        return advertising;
    }

    public void setAdvertising(String advertising) {
        this.advertising = advertising;
    }

    public int getRssiValue() {
        return rssiValue;
    }

    public void setRssiValue(int rssiValue) {
        this.rssiValue = rssiValue;
    }

    public boolean isRssiFlag() {
        return rssiFlag;
    }

    public void setRssiFlag(boolean rssiFlag) {
        this.rssiFlag = rssiFlag;
    }

    public boolean isOnlyFavourite() {
        return onlyFavourite;
    }

    public void setOnlyFavourite(boolean onlyFavourite) {
        this.onlyFavourite = onlyFavourite;
    }

    public List<BleFormat> getBleFormats() {
        return bleFormats;
    }

    public void setBleFormats(List<BleFormat> bleFormats) {
        this.bleFormats = bleFormats;
    }

    public String getFilterName() {
        return filterName;
    }

    public void setFilterName(String filterName) {
        this.filterName = filterName;
    }

    public boolean isOnlyConnectable() {
        return onlyConnectable;
    }

    public void setOnlyConnectable(boolean onlyConnectable) {
        this.onlyConnectable = onlyConnectable;
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Utils/FilterDeviceParams.kt
package com.siliconlabs.bledemo.utils

import android.content.Context
import com.siliconlabs.bledemo.R
import com.siliconlabs.bledemo.bluetooth.beacon_utils.BleFormat
import kotlinx.android.synthetic.main.fragment_filter.*

data class FilterDeviceParams(
        val name: String?,
        val rssiValue: Int,
        val isRssiFlag: Boolean,
        val bleFormats: List<BleFormat>,
        val isOnlyFavourite: Boolean,
        val isOnlyConnectable: Boolean,
        val isOnlyBonded: Boolean
) {

    val isEmpty: Boolean
        get() = name == null
                && !isRssiFlag
                && !isOnlyFavourite
                && !isOnlyConnectable
                && !isOnlyBonded
                && bleFormats.isEmpty()

    fun buildDescription(context: Context?) : String? {
        if (context == null) return null

        val description = StringBuilder().apply {
            if (name?.isNotBlank() == true) append("\"$name\", ")
            if (isRssiFlag) append("> $rssiValue dBm, ")
            if (bleFormats.isNotEmpty()) {
                bleFormats.forEach { append(context.getString(it.nameResId)).append(", ") }
            }
            if (isOnlyFavourite) {
                append(context.getString(R.string.only_favorites)).append(", ")
            }
            if (isOnlyConnectable) {
                append(context.getString(R.string.only_connectible)).append(", ")
            }
            if (isOnlyBonded) {
                append(context.getString(R.string.only_bonded)).append(", ")
            }
        }.toString()

        return if (description.isEmpty()) null else description.substring(0, description.count() - 2)
    }

}
